### PR TITLE
Add vitests

### PR DESCRIPTION
**Pull request by ribru17**
Add vitest tests that are compatible with webpack (and thus our project)

This also removes all babel/jest code from our project to prevent unneeded bloat

### CODE REVIEW

File: shared_types/exercise.ts
Comment (annaellebgt) [inline shared_types/exercise.ts:1]: you are removing important changes I had done !!

File: shared_types/scamCategory.ts
Comment (annaellebgt) [inline shared_types/scamCategory.ts:1]: same here, you are not supposed to remove the export

File: shared_types/user.ts
Comment (annaellebgt) [inline shared_types/user.ts:1]: you should leave the overallScore and the graduated fields

**Review by annaellebgt — CHANGES_REQUESTED** [review]
I think there is one commit that I did before to add the missing attributes that you are reverting #40

**Comment by ribru17** [comment]
Yeah you are right- I too am realizing now that we cannot use `git revert` because merge commits are in the history :cry: So  even though I `git revert`ed only the last 5 commits, much more changes were undone. I will have to take the painful, manual approach. Marking as draft for now

**Pull request closed without merging**